pingcap/tidb
# config,go.mod: support timeout for log writting

### What problem does this PR solve?


Issue Number: close #35007

Problem Summary:

### What is changed and how it works?

~~Support nonblock configuration for writing log.~~

Support timeout configure for log writing
When tidb block a long time on log writing, let it crash instead of hang
See the actual changes in the log lib https://github.com/pingcap/log/pull/36/files

### Check List

Tests 


1. Get a USB flash disk, mount it to `/home/genius/mnt/xxx`
2. Configure the tidb-log / slow-query-log to that dir
3. Run `./bin/tidb-server -config config.toml.example`
4. Plug out the USB flash disk
5. Check tidb process does not hang


Side effects


Documentation

Need to update the document about the new configuration

### Release note



Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```
